### PR TITLE
feature/api-docs

### PR DESCRIPTION
**Pull request by dabeeeenster**
Implement OpenAPI plugin for docusaurus so we can define our Edge API endpoints

### CODE REVIEW

**Comment by dabeeeenster** [comment]
Is it possible to just have a sample json document as the response documentation?

File: api_spec/edge_api.json [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+                    },
+                    "type": {
+                        "title": "Type",
+                        "type": "string",
+                        "maxLength": 50
+                    }
+                }
+            },
+            "createBodyRequest": {
+                "title": "object",
```
Comment (matthewelwell) [inline api_spec/edge_api.json:174]: I think the title here should be something more descriptive? `"Identities Request Data"` for example.

File: api_spec/edge_api.json [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+                        "minLength": 1
+                    },
+                    "type": {
+                        "title": "Type",
+                        "type": "string",
+                        "maxLength": 50
+                    }
+                }
+            },
+            "createBodyRequest": {
```
Comment (matthewelwell) [inline api_spec/edge_api.json:173]: Can we change the name of this to something specific to the `Identities` endpoint? `IdentitiesRequestData` perhaps?

File: api_spec/edge_api.json [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+                "title": "object",
+                "required": true,
+                "type": "object",
+                "properties": {
+                    "Identifier": {
```
Comment (matthewelwell) [inline api_spec/edge_api.json:178]: I think you've got the object key / title attribute the wrong way around. The object key should be the attribute that we expect the user to provide. The title is a user friendly name for the attribute. For example, this should be: 

```
"identifier": {
    "title": "Identifier"
    ...
}
```

Similarly for the traits, we should have something like: 

```json
"traits": {
    "title": "Traits"
    ...
}
```

File: api_spec/edge_api.json [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+                        "maxLength": 50
+                    },
+                    "Traits": {
+                        "title": "Traits",
+                        "type": "object",
+                        "properties": {
+                            "Trait key": {
+                                "title": "trait_key",
+                                "type": "string",
+                                "maxLength": 50
```
Comment (matthewelwell) [inline api_spec/edge_api.json:190]: I don't think this `maxLength` here is correct. There probably isn't a max length on these to be honest. You can just leave this attribute off.

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**